datafusion/tests/sql.rs

```diff
@@ -2124,6 +2124,49 @@ async fn cross_join_unbalanced() {
     );
 }
 
+#[tokio::test]
+async fn test_join_timestamp() -> Result<()> {
+    let mut ctx = ExecutionContext::new();
+
+    // register time table
```

Comment:
Also, the function `table_with_timestamps` might help writing tests for other timestamp units: 
 https://github.com/apache/arrow-datafusion/blob/master/datafusion/src/test/mod.rs#L194